src/chatd.cpp

```diff
@@ -4672,6 +4672,11 @@ void Chat::msgIncomingAfterDecrypt(bool isNew, bool isLocal, Message& msg, Idx i
         CALL_LISTENER(onMsgOrderVerificationFail, msg, idx, "A message with that backrefId "+std::to_string(msg.backRefId)+" already exists");
     }
 
+    if (isPublic() && msg.userid != mChatdClient.mMyHandle)
```

Comment:
If there is a way of checking the attributes are already fetched, perhaps we can skip the request, considering this piece of code will be executed per message.

Reply:
I think the request would be generated the first time only. 
In method `UserAttrCache::getAttr(uint64_t userHandle, unsigned type, void* userp, UserAttrReqCbFunc cb, bool oneShot, bool fetch, uint64_t ph)` we would have an element in the map and item.pending would be different form kCacheFetchNewPending and kCacheNotFetchUntilUse, then we call directly to the callback.
Although request is not launched, we make many unnecessary method calls. Implement a method to check if fetch is necessary it would be ok

Reply:
Changed